dbus/xdg.c

```diff
@@ -240,6 +240,16 @@ static int handle_notify(sd_bus_message *msg, void *data,
 			// will win over app_icon if provided.
 			free(notif->app_icon);
 			notif->app_icon = strdup(image_path);
+		} else if (strcmp(hint, "synchronous") == 0 ||
+				strcmp(hint, "private-synchronous") == 0 ||
+				strcmp(hint, "x-canonical-private-synchronous") == 0 ||
+				strcmp(hint, "x-dunst-stack-tag") == 0) {
```

Comment:
Hm. Do we really need to support all of these? `private-synchronous` and `x-canonical-private-synchronous` seem a little alien. `synchronous` seems like it's polluting the hint namespace. I get that this isn't standardized, but do we have an idea how much these are used in practice?

Reply:
The list was copied from those supported by Dunst; I thought it was apt to spare people migrating over to mako from modifying their existing scripts.

`x-dunst-stack-tag` and `private-synchronous` are used by Dunst alone. `x-canonical-private-synchronous` is the only hint officially supported by NotifyOSD, but `synchronous` is [supported for compatibility with older applications](https://bazaar.launchpad.net/~indicator-applet-developers/notify-osd/trunk/view/head:/src/stack.c#L696).

`private-synchronous` appears to be intended as a notification-daemon independent alternative.

How about going with `x-canonical-private-synchronous` for compatibility with newer applications using Canonical's "official" private API, and `private-synchronous` for general usage by scripts?

Reply:
> private-synchronous appears to be intended as a notification daemon agnostic alternative.

It seems like notify-osd was using `private-synchronous` before they were made aware that non-standard keys must be vendored. Ref: https://bugs.launchpad.net/ubuntu/+source/notify-osd/+bug/340817

> How about going with x-canonical-private-synchronous for compatibility with newer applications using Canonical's "official" private API, and private-synchronous as an agnostic alternative for scripts?

I'd rather not recommend `private-synchronous` at all, since it's been removed from the original implementation in notify-osd. Also the "private" prefix doesn't really make sense (it probably means "private to Canonical").

I'm fine with supporting `x-dunst-stack-tag` and `x-canonical-private-synchronous`.

Also ref the dunst PR: https://github.com/dunst-project/dunst/pull/552

Reply:
Hm, there's also a difference between the cap name and the hint name: https://github.com/dunst-project/dunst/pull/552#issuecomment-433630749

Reply:
Adjusted to only support `x-canonical-private-synchronous` and `x-dunst-stack-tag`. Kept the capabilities as they are to `private-synchronous` and `x-canonical-private-synchronous` since it's difficult to tell exactly what's expected. Seems reasonable to me to go with what Canonical supports and most applications are using.

Reply:
It seems like the capabilities match the hint names, or at least dunst assumes so. Can we remove `private-synchronous` from the caps, and add `x-dunst-stack-tag`?